### PR TITLE
DPC-326: As an internal user, I want to filter organizations

### PR DESCRIPTION
**Pull request by Sun-Mountain**
**Why**

The filter for organizations was broken and internal users would not be able to more succinctly search for specific organizations.

Screen Shot 2020-06-16 at 12 19 43 PM

**What Changed**

Screen Shot 2020-06-17 at 4 59 22 PM

- Added filter modal for the organization dashboard allowing the internal user to filter by:
   -  Whether or not the org is enabled in the API
   - Created before/after dates
   - Organization type

**Choices Made**

- Combined user search and organization search into a "Base Search" service file

**Tickets closed**:

[DPC-326](https://jira.cms.gov/browse/DPC-326)

**Future Work**

N/A

**Checklist**

- [x] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [\] Swagger documentation has been updated
- [\] FHIR documentation has been updated
- [\] Any required dpc-ops changes have a PR submitted and mentioned in this ticket
- [\] Any manual migration steps are documented, scripts written (where applicable), and tested
- [\] Before merging, any required dpc-ops changes have been approved and merged into master of the dpc-ops repo

### CODE REVIEW

File: dpc-web/app/controllers/internal/organizations_controller.rb [hunk 1]
```diff
@@ -5,22 +5,9 @@ class OrganizationsController < ApplicationController
...
-      if keyword_param[:keyword].present?
-        keyword = "%#{keyword_param[:keyword].downcase}%"
-        scope = scope.where(
-          'LOWER(name) LIKE :keyword', keyword: keyword
-        )
-      end
+      results = BaseSearch.new(params: params, scope: params[:org_type]).results
 
-      @organizations = scope.page params[:page]
+      @organizations = results.page params[:page]
```
Comment (dhgreene) [inline dpc-web/app/controllers/internal/organizations_controller.rb:10]: Do we need to filter here which parameters we accept, to be consistent with #843 ?

Comment (Sun-Mountain) [reply dpc-web/app/controllers/internal/organizations_controller.rb:10]: This is just referencing pagination when listing over a certain amount of organizations. I can move it to private like #843

Comment (Sun-Mountain) [reply dpc-web/app/controllers/internal/organizations_controller.rb:10]: I could put `results.page page[:params]` in a private method like `org_page_params(results)`

Comment (Sun-Mountain) [reply dpc-web/app/controllers/internal/organizations_controller.rb]: I just did it :)

**Pull request merged**